test/gui/shared/scripts/pageObjects/PublicLinkDialog.py

```diff
@@ -137,17 +159,45 @@ def setExpirationDate(self, context, publicLinkName, resource):
         squish.nativeType(expDate.day)
         squish.nativeType(expYear)
         squish.nativeType("<Return>")
+
+        actualDate = squish.waitForObjectExists(self.EXPIRATION_DATE_FIELD).displayText
+        expectedDate = f"{expDate.month}/{expDate.day}/{expYear}"
+        if not actualDate == expectedDate:
+            # retry with workaround
+            self.setExpirationDateWithWorkaround(expYear, expDate.month, expDate.day)
+
         squish.testSettings.silentVerifications = True
         squish.waitFor(
-            lambda: (test.xvp("publicLinkExpirationProgressIndicatorInvisible"))
+            lambda: (test.vp("publicLinkExpirationProgressIndicatorInvisible"))
         )
-        waitFor(lambda: (test.vp("publicLinkExpirationProgressIndicatorInvisible")))
         squish.testSettings.silentVerifications = False
         test.compare(
             str(squish.waitForObjectExists(self.EXPIRATION_DATE_FIELD).displayText),
-            str(expDate.month) + "/" + str(expDate.day) + "/" + str(expYear),
+            str(expectedDate),
+        )
+
+    # This workaround is needed because the above function 'setExpirationDate'
+    # will not work while creating new public link
+    # See for more details: https://github.com/owncloud/client/issues/9218
+    def setExpirationDateWithWorkaround(self, year, month, day):
+        squish.mouseClick(
+            squish.waitForObject(self.EXPIRATION_DATE_FIELD),
+            0,
+            0,
+            squish.Qt.NoModifier,
+            squish.Qt.LeftButton,
         )
 
+        squish.nativeType("<Ctrl+Right>")
+        squish.nativeType("<Ctrl+Right>")
+        squish.nativeType(year)
+        squish.nativeType("<Ctrl+Left>")
+        squish.nativeType(day)
+        squish.nativeType("<Ctrl+Left>")
+        squish.nativeType("<Ctrl+Left>")
+        squish.nativeType(month)
+        squish.nativeType("<Return>")
+
```

Comment:
Can't we do this in one line? 🤔 the documentation is `nativeType(keys);` so can you try if you can send all the keys in a single call? such as `squish.nativeType("<Ctrl+Right>", "<Ctrl+Right>", year)`. I am not sure if it works.

Also, can you add some comment here.

Reply:
`nativeType` seems to accept only one argument. So cannot send multiple keys at once.